src/utils/SpotifyAPI/SpotifyAPI.js

```diff
@@ -0,0 +1,179 @@
+import { checkResponse } from "../api";
+const clientId = "774dc384c15942b2a0637338f1e99147";
+const clientSecret = "431e5d12c78344e3a1a58747820c883c";
+const market = "US";
+
+export const getToken = () => {
+  let authParameters = {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/x-www-form-urlencoded",
+    },
+    body:
+      "grant_type=client_credentials&client_id=" +
+      clientId +
+      "&client_secret=" +
+      clientSecret,
+  };
+  const authToken = fetch(
+    "https://accounts.spotify.com/api/token",
+    authParameters
+  ).then((res) => {
+    return checkResponse(res);
```

Comment:
**Could be improved**

You can make a special method (function) for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(checkResponse)
  }

```

And now you can safely replace `fetch` with `request` and remove repeated checking. And you don’t need to change anything else.